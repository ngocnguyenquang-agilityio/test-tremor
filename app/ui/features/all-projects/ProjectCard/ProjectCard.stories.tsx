import type { Meta, StoryObj } from "@storybook/react";

// Components
import ProjectCard from "./ProjectCard";

//Constans
import { ITEM_ACTION_PROJECT } from "@/constants";

//Mocks
import { PROJECT_DATA } from "@/mocks/project";

const meta = {
  title: "Components/ProjectCard",
  component: ProjectCard,
  tags: ["autodocs"],
} as Meta<typeof ProjectCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ProjectCardDefault: Story = {
  render: () => (
    <ProjectCard projectData={PROJECT_DATA[0]} actions={ITEM_ACTION_PROJECT} />
  ),
};
