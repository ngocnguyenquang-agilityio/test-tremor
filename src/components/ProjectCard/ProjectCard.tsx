// Components
import { Card, Text, Flex, Title, Button } from "@tremor/react";
import Avatar from "../../components/Avatar/Avatar";
import SlackIcon from "../../components/Icons/SlackIcon";

//Types
import { Project } from "../../types/project";

//Constans
import {
  ITEM_ACTION_PROJECT,
  URL_AVATAR_PROJECT,
} from "../../constants/commons";

type AcionCard = {
  key: string;
  label: string;
};
interface IProjectCard {
  projectData: Project;
  actions: AcionCard[];
  projectId: string;
  onToggleAction: (project: Project) => void;
  onActionProject: (project: Project, action: string) => void;
  isOpenAction: boolean;
}

const data = {
  id: "1",
  icon: <SlackIcon />,
  avatars: URL_AVATAR_PROJECT,
  title: "Slack Bot",
  date: "02.03.22",
  participant: 5,
  description:
    "If everything I did failed - which it doesn't, I think that it actually succeeds.",
};

const ProjectCard = ({
  projectData = data,
  actions = ITEM_ACTION_PROJECT,
  isOpenAction = false,
  projectId = "1",
  onToggleAction,
  onActionProject,
}: IProjectCard): JSX.Element => {
  const { icon, title, date, participant, description, avatars, id } =
    projectData;

  const openActionProject = isOpenAction && id === projectId;

  const handleItemActionProject = (
    event: React.MouseEvent<HTMLElement>,
    project: Project,
    action: string,
  ) => {
    event.preventDefault();
    onActionProject(project, action);
  };

  return (
    <div className="antialiased items-center justify-between py-1">
      <div className="flex items-center">
        <Card className="mx-auto px-4 py-1 ring-0 max-w-full lg:max-w-[356px] border-none relative mt-[40px] rounded-xl shadow-md">
          <Flex className="absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-[linear-gradient(195deg,#42424a,#191919)] justify-center rounded-xl">
            {icon}
          </Flex>
          <Flex className="pl-[90px] mb-6 relative">
            <Flex className="flex-col items-start justify-start ">
              <Title className="text-xl font-bold text-base leading-5">
                {title}
              </Title>
              <Flex className="items-start justify-start ml-[10px]">
                {avatars.map(avatar => (
                  <Avatar
                    key={avatar.key}
                    alt="Avatar extra small"
                    className="border-2 border-white border-solid ml-[-10px]"
                    height={20}
                    priority
                    src={avatar.src}
                    sizes="(max-width: 768px) 100vw, 33vw"
                    width={20}
                  />
                ))}
              </Flex>
            </Flex>
            <Flex className="flex-col w-auto justify-end">
              <Flex
                className="cursor-pointer flex-col w-[30px] h-[16px] justify-between"
                onClick={onToggleAction}>
                <Text className="w-[4px] h-[4px] rounded-full bg-[#7b809a]" />
                <Text className="w-[4px] h-[4px] rounded-full bg-[#7b809a]" />
                <Text className="w-[4px] h-[4px] rounded-full bg-[#7b809a]" />
              </Flex>
              {openActionProject && (
                <div className="absolute px-[18px] py-2 right-[26px] top-[5px] z-10 bg-white rounded-md shadow-md">
                  {actions.map(item => (
                    <Flex key={item.key} flex-col>
                      <Button
                        className="w-full justify-start text-tremor-content-title hover:text-tremor-content-title hover:bg-[#f0f2f5] hover:rounded-md px-1 py-[6px]"
                        variant="light"
                        onClick={(event: React.MouseEvent<HTMLButtonElement>) =>
                          handleItemActionProject(
                            event,
                            projectData,
                            item.label,
                          )
                        }>
                        <Text className="font-normal text-sm text-[#7b809a]">
                          {item.label}
                        </Text>
                      </Button>
                    </Flex>
                  ))}
                </div>
              )}
            </Flex>
          </Flex>
          <Text className="text-md text-[#7b809a] my-4 font-light">
            {description}
          </Text>
          <div className="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4" />
          <Flex className="my-4">
            <div>
              <Text className="text-[#344767] font-semibold">
                {participant}
              </Text>
              <Text className="text-[#7b809a] font-normal">Participants</Text>
            </div>
            <div>
              <Text className="text-[#344767] font-semibold">{date}</Text>
              <Text className="text-[#7b809a] font-normal">Due date</Text>
            </div>
          </Flex>
        </Card>
      </div>
    </div>
  );
};

export default ProjectCard;
