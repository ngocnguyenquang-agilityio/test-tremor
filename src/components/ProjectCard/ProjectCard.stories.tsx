import type { Meta, StoryObj } from "@storybook/react";

// Components
import ProjectCard from "./ProjectCard";
import { SlackIcon } from "../../components/Icons/SlackIcon";

//Constans
import {
  ITEM_ACTION_PROJECT,
  URL_AVATAR_PROJECT,
} from "../../constants/commons";

export default { component: ProjectCard };
const meta = {
  title: "Components/ProjectCard",
  component: ProjectCard,
  tags: ["autodocs"],
} as Meta<typeof ProjectCard>;

type Story = StoryObj<typeof meta>;

export const ProjectCardDefault: Story = {
  render: () => (
    <ProjectCard
      icon={<SlackIcon />}
      title="Slack Bot"
      date="02.03.22"
      participant={5}
      description="If everything I did failed - which it doesn't, I think that it actually succeeds."
      avatars={URL_AVATAR_PROJECT}
      actions={ITEM_ACTION_PROJECT}
      isOpenAction={false}
      handletToggleAction={() => {}}
    />
  ),
};
