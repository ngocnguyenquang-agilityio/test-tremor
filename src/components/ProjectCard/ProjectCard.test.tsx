import { render } from "@testing-library/react";

// Components
import ProjectCard from "./ProjectCard";
import { SlackIcon } from "../../components/Icons/SlackIcon";

//Constans
import {
  ITEM_ACTION_PROJECT,
  URL_AVATAR_PROJECT,
} from "../../constants/commons";

describe("Testing ProjectCard component", () => {
  const propsDefault = {
    icon: <SlackIcon />,
    title: "Slack Bot",
    date: "02.03.22",
    participant: 5,
    description:
      "If everything I did failed - which it doesn't, I think that it actually succeeds.",
    avatars: URL_AVATAR_PROJECT,
    actions: ITEM_ACTION_PROJECT,
    isOpenAction: false,
    handletToggleAction: () => {},
  };

  it("Should match snapshot", () => {
    const component = render(<ProjectCard {...propsDefault} />);
    expect(component).toMatchSnapshot();
  });
});
