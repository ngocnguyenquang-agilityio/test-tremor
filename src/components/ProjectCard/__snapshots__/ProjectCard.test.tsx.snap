// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProjectCard component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="antialiased items-center justify-between py-1"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 mx-auto px-4 py-1 ring-0 max-w-[356px] border-none relative mt-[40px] rounded-xl shadow-md"
          >
            <div
              class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-[linear-gradient(195deg,#42424a,#191919)] justify-center rounded-xl"
            >
              <svg
                height="150px"
                version="1.1"
                viewBox="0 0 177 150"
                width="177px"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  id="Logos"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill-rule="nonzero"
                    id="slack-new-logo"
                    transform="translate(35.000000, 22.000000)"
                  >
                    <path
                      d="M39.2803996,-1.77635684e-15 C33.3581411,0.00436871997 28.5651823,4.80122324 28.5695565,10.7121014 C28.5651823,16.6229795 33.3625182,21.419834 39.2847767,21.4242027 L49.999997,21.4242027 L49.999997,10.7164701 C50.0043711,4.80559196 45.2070353,0.00873743993 39.2803996,-1.77635684e-15 C39.2847767,-1.77635684e-15 39.2847767,-1.77635684e-15 39.2803996,-1.77635684e-15 M39.2803996,28.5714286 L10.7152412,28.5714286 C4.7929826,28.5757973 -0.00435323957,33.3726518 1.19295144e-05,39.2835299 C-0.0087303708,45.194408 4.78860547,49.9912626 10.710864,50 L39.2803996,50 C45.2026582,49.9956313 49.999994,45.1987768 49.9956199,39.2878986 C49.999994,33.3726518 45.2026582,28.5757973 39.2803996,28.5714286 Z"
                      fill="#36C5F0"
                      id="Shape"
                    />
                    <path
                      d="M106.999997,39.2835299 C107.004372,33.3726518 102.206616,28.5757973 96.2838386,28.5714286 C90.3610616,28.5757973 85.5633057,33.3726518 85.5676802,39.2835299 L85.5676802,50 L96.2838386,50 C102.206616,49.9956313 107.004372,45.1987768 106.999997,39.2835299 Z M78.4279602,39.2835299 L78.4279602,10.7121014 C78.4323347,4.80559196 73.6389563,0.00873743993 67.7161793,7.10542736e-15 C61.7934022,0.00436871997 56.9956464,4.80122324 57.0000119,10.7121014 L57.0000119,39.2835299 C56.9912689,45.194408 61.7890247,49.9912626 67.7118018,50 C73.6345788,49.9956313 78.4323347,45.1987768 78.4279602,39.2835299 Z"
                      fill="#2EB67D"
                      id="Shape"
                    />
                    <path
                      d="M67.7152233,107 C73.6374818,106.995632 78.4348177,102.199196 78.4304435,96.2888345 C78.4348177,90.3784728 73.6374818,85.5820374 67.7152233,85.5776691 L57.000003,85.5776691 L57.000003,96.2888345 C56.9956289,102.194828 61.7929647,106.991263 67.7152233,107 Z M67.7152233,78.4266993 L96.2847588,78.4266993 C102.207017,78.4223309 107.004353,73.6258955 106.999988,67.7155338 C107.00873,61.8051721 102.211395,57.0087367 96.289136,57 L67.7196004,57 C61.7973418,57.0043683 57.000006,61.8008038 57.0043801,67.7111655 C57.000006,73.6258955 61.7929647,78.4223309 67.7152233,78.4266993 L67.7152233,78.4266993 Z"
                      fill="#ECB22E"
                      id="Shape"
                    />
                    <path
                      d="M2.98908926e-06,67.7121014 C-0.00437153219,73.6229795 4.79338431,78.419834 10.7161614,78.4242027 C16.6389384,78.419834 21.4366943,73.6229795 21.4323198,67.7121014 L21.4323198,57 L10.7161614,57 C4.79338431,57.0043687 -0.00437153219,61.8012232 2.98908926e-06,67.7121014 Z M28.5720309,67.7121014 L28.5720309,96.2835299 C28.5632878,102.194408 33.3610437,106.991263 39.2838207,107 C45.2065978,106.995631 50.0043536,102.198777 49.9999881,96.2878986 L49.9999881,67.7208388 C50.0087311,61.8099607 45.2109753,57.0131062 39.2881982,57.0043687 C33.3610437,57.0043687 28.5676653,61.8012232 28.5720309,67.7121014 C28.5720309,67.7121014 28.5720309,67.7164701 28.5720309,67.7121014"
                      fill="#E01E5A"
                      id="Shape"
                    />
                  </g>
                </g>
              </svg>
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center pl-[90px] mb-6 relative"
            >
              <div
                class="tremor-Flex-root flex w-full flex-col items-start justify-start"
              >
                <p
                  class="text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis font-bold text-base"
                >
                  Slack Bot
                </p>
                <div
                  class="tremor-Flex-root flex w-full flex-row items-start justify-start ml-[10px]"
                >
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                  <div
                    class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                  >
                    <img
                      alt="Avatar extra small"
                      data-nimg="1"
                      decoding="async"
                      fetchpriority="high"
                      height="20"
                      sizes="(max-width: 768px) 100vw, 33vw"
                      src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                      style="color: transparent;"
                      width="20"
                    />
                  </div>
                </div>
              </div>
              <div
                class="tremor-Flex-root flex items-center flex-col w-auto justify-end"
              >
                <div
                  class="tremor-Flex-root flex items-center cursor-pointer flex-col w-[30px] h-[16px] justify-between"
                >
                  <p
                    class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                  />
                  <p
                    class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                  />
                  <p
                    class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                  />
                </div>
              </div>
            </div>
            <p
              class="text-tremor-default dark:text-dark-tremor-content text-md text-[#7b809a] my-4 font-light"
            >
              If everything I did failed - which it doesn't, I think that it actually succeeds.
            </p>
            <div
              class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4"
            />
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center my-4"
            >
              <div>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#344767] font-semibold"
                >
                  5
                </p>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#7b809a] font-normal"
                >
                  Participants
                </p>
              </div>
              <div>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#344767] font-semibold"
                >
                  02.03.22
                </p>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#7b809a] font-normal"
                >
                  Due date
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="antialiased items-center justify-between py-1"
    >
      <div
        class="flex items-center"
      >
        <div
          class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 mx-auto px-4 py-1 ring-0 max-w-[356px] border-none relative mt-[40px] rounded-xl shadow-md"
        >
          <div
            class="tremor-Flex-root flex flex-row items-center absolute top-[-22px] left-40px w-[74px] h-[74px] p-1 bg-[linear-gradient(195deg,#42424a,#191919)] justify-center rounded-xl"
          >
            <svg
              height="150px"
              version="1.1"
              viewBox="0 0 177 150"
              width="177px"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                id="Logos"
                stroke="none"
                stroke-width="1"
              >
                <g
                  fill-rule="nonzero"
                  id="slack-new-logo"
                  transform="translate(35.000000, 22.000000)"
                >
                  <path
                    d="M39.2803996,-1.77635684e-15 C33.3581411,0.00436871997 28.5651823,4.80122324 28.5695565,10.7121014 C28.5651823,16.6229795 33.3625182,21.419834 39.2847767,21.4242027 L49.999997,21.4242027 L49.999997,10.7164701 C50.0043711,4.80559196 45.2070353,0.00873743993 39.2803996,-1.77635684e-15 C39.2847767,-1.77635684e-15 39.2847767,-1.77635684e-15 39.2803996,-1.77635684e-15 M39.2803996,28.5714286 L10.7152412,28.5714286 C4.7929826,28.5757973 -0.00435323957,33.3726518 1.19295144e-05,39.2835299 C-0.0087303708,45.194408 4.78860547,49.9912626 10.710864,50 L39.2803996,50 C45.2026582,49.9956313 49.999994,45.1987768 49.9956199,39.2878986 C49.999994,33.3726518 45.2026582,28.5757973 39.2803996,28.5714286 Z"
                    fill="#36C5F0"
                    id="Shape"
                  />
                  <path
                    d="M106.999997,39.2835299 C107.004372,33.3726518 102.206616,28.5757973 96.2838386,28.5714286 C90.3610616,28.5757973 85.5633057,33.3726518 85.5676802,39.2835299 L85.5676802,50 L96.2838386,50 C102.206616,49.9956313 107.004372,45.1987768 106.999997,39.2835299 Z M78.4279602,39.2835299 L78.4279602,10.7121014 C78.4323347,4.80559196 73.6389563,0.00873743993 67.7161793,7.10542736e-15 C61.7934022,0.00436871997 56.9956464,4.80122324 57.0000119,10.7121014 L57.0000119,39.2835299 C56.9912689,45.194408 61.7890247,49.9912626 67.7118018,50 C73.6345788,49.9956313 78.4323347,45.1987768 78.4279602,39.2835299 Z"
                    fill="#2EB67D"
                    id="Shape"
                  />
                  <path
                    d="M67.7152233,107 C73.6374818,106.995632 78.4348177,102.199196 78.4304435,96.2888345 C78.4348177,90.3784728 73.6374818,85.5820374 67.7152233,85.5776691 L57.000003,85.5776691 L57.000003,96.2888345 C56.9956289,102.194828 61.7929647,106.991263 67.7152233,107 Z M67.7152233,78.4266993 L96.2847588,78.4266993 C102.207017,78.4223309 107.004353,73.6258955 106.999988,67.7155338 C107.00873,61.8051721 102.211395,57.0087367 96.289136,57 L67.7196004,57 C61.7973418,57.0043683 57.000006,61.8008038 57.0043801,67.7111655 C57.000006,73.6258955 61.7929647,78.4223309 67.7152233,78.4266993 L67.7152233,78.4266993 Z"
                    fill="#ECB22E"
                    id="Shape"
                  />
                  <path
                    d="M2.98908926e-06,67.7121014 C-0.00437153219,73.6229795 4.79338431,78.419834 10.7161614,78.4242027 C16.6389384,78.419834 21.4366943,73.6229795 21.4323198,67.7121014 L21.4323198,57 L10.7161614,57 C4.79338431,57.0043687 -0.00437153219,61.8012232 2.98908926e-06,67.7121014 Z M28.5720309,67.7121014 L28.5720309,96.2835299 C28.5632878,102.194408 33.3610437,106.991263 39.2838207,107 C45.2065978,106.995631 50.0043536,102.198777 49.9999881,96.2878986 L49.9999881,67.7208388 C50.0087311,61.8099607 45.2109753,57.0131062 39.2881982,57.0043687 C33.3610437,57.0043687 28.5676653,61.8012232 28.5720309,67.7121014 C28.5720309,67.7121014 28.5720309,67.7164701 28.5720309,67.7121014"
                    fill="#E01E5A"
                    id="Shape"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center pl-[90px] mb-6 relative"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col items-start justify-start"
            >
              <p
                class="text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis font-bold text-base"
              >
                Slack Bot
              </p>
              <div
                class="tremor-Flex-root flex w-full flex-row items-start justify-start ml-[10px]"
              >
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
                <div
                  class="inline-flex rounded-full overflow-hidden border-2 border-white border-solid ml-[-10px]"
                >
                  <img
                    alt="Avatar extra small"
                    data-nimg="1"
                    decoding="async"
                    fetchpriority="high"
                    height="20"
                    sizes="(max-width: 768px) 100vw, 33vw"
                    src="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
                    style="color: transparent;"
                    width="20"
                  />
                </div>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex items-center flex-col w-auto justify-end"
            >
              <div
                class="tremor-Flex-root flex items-center cursor-pointer flex-col w-[30px] h-[16px] justify-between"
              >
                <p
                  class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                />
                <p
                  class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                />
                <p
                  class="text-tremor-default text-tremor-content dark:text-dark-tremor-content w-[4px] h-[4px] rounded-full bg-[#7b809a]"
                />
              </div>
            </div>
          </div>
          <p
            class="text-tremor-default dark:text-dark-tremor-content text-md text-[#7b809a] my-4 font-light"
          >
            If everything I did failed - which it doesn't, I think that it actually succeeds.
          </p>
          <div
            class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4"
          />
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center my-4"
          >
            <div>
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#344767] font-semibold"
              >
                5
              </p>
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#7b809a] font-normal"
              >
                Participants
              </p>
            </div>
            <div>
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#344767] font-semibold"
              >
                02.03.22
              </p>
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#7b809a] font-normal"
              >
                Due date
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
